flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -493,8 +493,14 @@ public static Path compressDirectory(Path directory, Path target) throws IOExcep
 		FileSystem sourceFs = directory.getFileSystem();
 		FileSystem targetFs = target.getFileSystem();
 
+		Path absolutePath;
+		if (!directory.isAbsolute()) {
+			absolutePath = new Path(sourceFs.getWorkingDirectory(), directory);
+		} else {
+			absolutePath = directory;
+		}
```

Comment:
Could you please extract to a public method in FileUtils? This seems useful enough on its own.

I was even thinking of adding it directly to path, but at this point, we shouldn't change public API anymore.